### PR TITLE
Update model-binding.md /patch

### PR DESCRIPTION
**Pull request by Rick-Anderson**
[Internal review URL](https://review.docs.microsoft.com/en-us/aspnet/core/mvc/models/model-binding?view=aspnetcore-7.0&branch=pr-en-us-26975#simple-and-complex-types)

@brunolins16 we never define simple and complex. This is just a start, can you help me define it? Should also talk about recursive binding.

From [your comment](https://github.com/dotnet/AspNetCore.Docs/pull/26897#issuecomment-1242506679)
> Maybe also include something related to the fact that it is not a complex model binding (recursive as MVC) and instead it will just expand the `properties` are handler `parameters`.

### CODE REVIEW

File: aspnetcore/mvc/models/model-binding.md [hunk 1]
```diff
@@ -23,6 +23,10 @@ Controllers and Razor pages work with data that comes from HTTP requests. For ex
 * Converts string data to .NET types.
 * Updates properties of complex types.
 
+## Simple and complex types
```
Comment (Rick-Anderson) [inline aspnetcore/mvc/models/model-binding.md:26]:
```suggestion
### Simple and complex types
```

File: aspnetcore/mvc/models/model-binding.md [hunk 2]
```diff
@@ -23,6 +23,10 @@ Controllers and Razor pages work with data that comes from HTTP requests. For ex
 * Converts string data to .NET types.
 * Updates properties of complex types.
 
+### Simple and complex types
+
+A simple type is one of the predefined `struct` types such as `int`, 'char`, `float`, `double`, and `bool`. For a complete list, see [Simple types](/dotnet/csharp/language-reference/language-specification/types#835-simple-types). A complex type is a type that is not a simple type, such as a `class` or [`record`](/dotnet/csharp/language-reference/builtin-types/record).
```
Comment (Rick-Anderson) [inline aspnetcore/mvc/models/model-binding.md:28]:
```suggestion
A simple type is one of the predefined `struct` types such as `int`, 'char`, `float`, `double`, and `bool`. For a complete list, see [Simple types](/dotnet/csharp/language-reference/language-specification/types#835-simple-types). A complex type is a type that is not a simple type, such as a `class` or  [`record`](/dotnet/csharp/language-reference/builtin-types/record).
```

File: aspnetcore/mvc/models/model-binding.md [hunk 3]
```diff
@@ -23,6 +23,10 @@ Controllers and Razor pages work with data that comes from HTTP requests. For ex
 * Converts string data to .NET types.
 * Updates properties of complex types.
 
+### Simple and complex types
+
+A simple type is one of the predefined `struct` types such as `int`, 'char`, `float`, `double`, and `bool`. For a complete list, see [Simple types](/dotnet/csharp/language-reference/language-specification/types#835-simple-types). A complex type is a type that is not a simple type, such as a `class` or  `record`](/dotnet/csharp/language-reference/builtin-types/record).
```
Comment (Rick-Anderson) [inline aspnetcore/mvc/models/model-binding.md:28]:
```suggestion
A simple type is one of the predefined `struct` types such as `int`, 'char`, `float`, `double`, and `bool`. For a complete list, see [Simple types](/dotnet/csharp/language-reference/language-specification/types#835-simple-types). A complex type is a type that is not a simple type, such as a `class` or  [`record`](/dotnet/csharp/language-reference/builtin-types/record).
```

File: aspnetcore/mvc/models/model-binding.md [hunk 4]
```diff
@@ -23,6 +23,10 @@ Controllers and Razor pages work with data that comes from HTTP requests. For ex
 * Converts string data to .NET types.
 * Updates properties of complex types.
 
+### Simple and complex types
+
+A simple type is one of the predefined `struct` types such as `int`, 'char`, `float`, `double`, and `bool`. For a complete list, see [Simple types](/dotnet/csharp/language-reference/language-specification/types#835-simple-types). A complex type is a type that is not a simple type, such as a `class` or  [`record`](/dotnet/csharp/language-reference/builtin-types/record).
```
Comment (Rick-Anderson) [inline aspnetcore/mvc/models/model-binding.md:28]:
```suggestion

A simple type is one of the predefined `struct` types such as `int`, `char`, `float`, `double`, and `bool`. For a complete list, see [Simple types](/dotnet/csharp/language-reference/language-specification/types#835-simple-types). A complex type is a type that is not a simple type, such as a `class` or  [`record`](/dotnet/csharp/language-reference/builtin-types/record).

```

**Comment by Rick-Anderson** [comment]
@brunolins16 we never define simple and complex. This is just a start, can you help me define it? Should also talk about recursive binding.

From [your comment](https://github.com/dotnet/AspNetCore.Docs/pull/26897#issuecomment-1242506679)
> Maybe also include something related to the fact that it is not a complex model binding (recursive as MVC) and instead it will just expand the `properties` are handler `parameters`.

File: aspnetcore/mvc/models/model-binding.md [hunk 5]
```diff
@@ -23,6 +23,12 @@ Controllers and Razor pages work with data that comes from HTTP requests. For ex
 * Converts string data to .NET types.
 * Updates properties of complex types.
 
+### Simple and complex types
+
+
+A simple type is one of the predefined `struct` types such as `int`, `char`, `float`, `double`, and `bool`. For a complete list, see [Simple types](/dotnet/csharp/language-reference/language-specification/types#835-simple-types). A complex type is a type that is not a simple type, such as a `class` or  [`record`](/dotnet/csharp/language-reference/builtin-types/record).
```
Comment (brunolins16) [inline aspnetcore/mvc/models/model-binding.md:29]: In MVC, simple types are basically defined as a type that contains a `TypeConverter` from string or TryParse method, that is a very different definition from the language

Comment (brunolins16) [reply aspnetcore/mvc/models/model-binding.md:29]: In this paragraph there is a briefly explanation https://learn.microsoft.com/en-us/aspnet/core/mvc/advanced/custom-model-binding?view=aspnetcore-6.0#model-binding-review

Comment (brunolins16) [reply aspnetcore/mvc/models/model-binding.md:29]: https://github.com/dotnet/AspNetCore.Docs/issues/27389

**Comment by Rick-Anderson** [comment]
Moved to #27415

**Pull request closed without merging**